mempool.go

```diff
@@ -1089,26 +1094,39 @@ func (mp *txMemPool) processOrphans(hash *btcwire.ShaHash) error {
 			enext = e.Next()
 			tx := e.Value.(*btcutil.Tx)
 
-			// Remove the orphan from the orphan pool.
+			// Remove the orphan from the orphan pool.  Current
+			// behavior requires that all saved orphans with
+			// a newly accepted parent are removed from the orphan
+			// pool and potentially added to the memory pool, but
+			// transactions which cannot be added to memory pool
+			// (including due to still being orphans) are expunged
+			// from the orphan pool.
+			//
+			// TODO(jrick): The above described behavior sounds
+			// like a bug, and I think we should investigate
+			// potentially moving orphans to the memory pool, but
+			// leaving them in the orphan pool if not all parent
+			// transactions are known yet.
 			orphanHash := tx.Sha()
 			mp.removeOrphan(orphanHash)
 
 			// Potentially accept the transaction into the
 			// transaction pool.
-			var isOrphan bool
-			err := mp.maybeAcceptTransaction(tx, &isOrphan, true, true)
+			missingParents, err := mp.maybeAcceptTransaction(tx,
+				true, true)
 			if err != nil {
 				return err
 			}
-
-			if !isOrphan {
-				// Generate the inventory vector and relay it.
-				iv := btcwire.NewInvVect(btcwire.InvTypeTx, tx.Sha())
-				mp.server.RelayInventory(iv)
-			} else {
-				mp.removeOrphan(orphanHash)
+			if len(missingParents) != 0 {
```

Comment:
The behavior has been changed here.  I'll have to review it more closely to see the implications, but I wanted to point out that this is a change in behavior.  Previously, it removed the orphan from the pool and pushed the orphan hash on to the process hashes list.  Now, it is doing neither of those things.